types/base.d.ts

```diff
@@ -120,5 +120,13 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 	context: any;
 }
 
+export interface Macro<T, I, E> {
+	(t: T, input: I, expected: E): void;
+	title? (providedTitle: string, input: I, expected: E): string;
```

Comment:
> Should we call this `title: string` instead of `providedTitle: string`? 

I would agree with @ivogabe I would lean against renaming, it could be confusing. Additionally `providedTitle` is how this is documented in the readme.

> I did a quick test and autocompletion doesn't show the argument name when implementing it like this

I've noticed autocomplete is not suggesting as well, though the type checker is validating correctly.
I'm open to suggestions on how to improve the typing to get autocompletion here.